src/core/AutoRest.Core/Model/Method.cs

```diff
@@ -268,5 +268,8 @@ public bool StructurallyEquals(Method other)
         [JsonIgnore]
         public virtual HashSet<string> LocallyUsedNames { get; } = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
+        [JsonIgnore]
+        public virtual bool IsXNullableReturnType { get { return (Responses.Keys.Any(key => Responses[key].Extensions?.Get<bool>("x-nullable") ?? true)); } }
```

Comment:
In https://github.com/Azure/autorest/pull/1804#issuecomment-278780237 it sounds like you check if *all* responses response types are marked as nullable, but this checks whether *one* response is. That's why I asked for more tests, we would almost have missed this. Please don't only add tests that test for the "good" behavior.